pandas/tseries/tools.py

```diff
@@ -273,6 +275,19 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     99   2000-04-09
     Length: 100, dtype: datetime64[ns]
 
+    Infer the format from the first entry
+
+    >>> pd.to_datetime(df.month + '/' +  df.day + '/' + df.year,
+                       infer_datetime_format=True)
```

Comment:
this is not valid code (as ``df.month`` are integers), do something like: 

``s = df.month.astype(str) + '/' +  df.day.astype(str) + '/' + df.year.astype(str)``

then infer on that